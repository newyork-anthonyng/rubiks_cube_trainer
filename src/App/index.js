import React, { Component } from "react";
import PropTypes from "prop-types";
import getRandomSolve from "rubiks-cross-trainer";
import createEmptyArray from "create-empty-array";

class App extends Component {
  constructor(props) {
    super(props);

    this.optionsArray = createEmptyArray(8);
    this.initialDifficulty = 1;

    this.state = {
      difficulty: this.initialDifficulty,
      randomSolve:
        props.initialScramble || getRandomSolve(this.initialDifficulty)
    };
  }

  getRandomSolve = () => {
    this.setState(prevState => ({
      randomSolve: getRandomSolve(prevState.difficulty)
    }));
  };

  handleSelectChange = e => {
    this.setState({
      difficulty: e.target.value
    });
  };

  render() {
    const containerStyle = {
      display: "flex",
      flexDirection: "column"
    };

    const headerStyle = {
      textAlign: "center"
    };

    const formStyle = {
      alignItems: "center",
      display: "flex",
      flexDirection: "column",
      justifyContent: "center"
    };

    const selectStyle = {
      marginBottom: "1rem"
    };

    return (
      <div style={containerStyle}>
        <h1 style={headerStyle}>{this.state.randomSolve}</h1>

        <div style={formStyle}>
          <div style={selectStyle}>
            <span>Number of moves to generate cross:</span>
            <select
              value={this.state.difficulty}
              onChange={this.handleSelectChange}
            >
              {this.optionsArray.map((_, index) => (
                <option key={index} value={index + 1}>
                  {index + 1}
                </option>
              ))}
            </select>
          </div>
          <button onClick={this.getRandomSolve}>Get random solve</button>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  initialScramble: PropTypes.string
};

export default App;
