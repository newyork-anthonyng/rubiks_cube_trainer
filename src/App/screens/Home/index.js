import React from "react";

const Home = () => <h1>Happy Cubing</h1>;

export default Home;
