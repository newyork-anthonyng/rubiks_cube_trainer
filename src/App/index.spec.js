import React from "react";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import { StaticRouter } from "react-router-dom";
import Home from "./screens/Home";
import App from "./";

it("should render correctly", () => {
  const wrapper = mount(
    <StaticRouter context={{}}>
      <App />
    </StaticRouter>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});
