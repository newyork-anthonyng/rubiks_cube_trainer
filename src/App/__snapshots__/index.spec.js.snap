// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <div>
        <nav>
          <ul>
            <li>
              <Link
                replace={false}
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/cross-trainer"
              >
                <a
                  href="/cross-trainer"
                  onClick={[Function]}
                >
                  Cross Trainer
                </a>
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/scramble"
              >
                <a
                  href="/scramble"
                  onClick={[Function]}
                >
                  Scramble
                </a>
              </Link>
            </li>
          </ul>
        </nav>
        <hr />
        <metronome />
        <hr />
        <Route
          component={[Function]}
          exact={true}
          path="/"
        >
          <Home
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            staticContext={Object {}}
          >
            <h1>
              Happy Cubing
            </h1>
          </Home>
        </Route>
        <Route
          path="/cross-trainer"
          render={[Function]}
        />
        <Route
          path="/scramble"
          render={[Function]}
        />
      </div>
    </App>
  </Router>
</StaticRouter>
`;
